.github/workflows/flow-linux-platforms.yml

```diff
@@ -59,34 +64,83 @@ on:
         description: 'Whether to run standalone tests'
         type: boolean
         default: true
+      redis-ref:
+        type: string
 
 jobs:
   get-required-envs:
     runs-on: ubuntu-latest
     outputs:
-      platform: ${{ steps.get_platforms.outputs.platform }}
+      platforms_arm: ${{ steps.get_platforms.outputs.arm-platforms }}
+      platforms_x86: ${{ steps.get_platforms.outputs.x86-platforms }}
+      include_arm: ${{ steps.get_platforms.outputs.arm-include }}
+      include_x86: ${{ steps.get_platforms.outputs.x86-include }}
     steps:
       - name: Get platform
         id: get_platforms
+        shell: python
         run: |
-          if [ "${{ inputs.platform }}" == "all" ]; then
-            echo "platform=${{ env.ALL_IMAGES }}" >> $GITHUB_OUTPUT
-          else
-            echo "platform=['${{ inputs.platform }}']" >> $GITHUB_OUTPUT
-          fi
+          import os
+
+          requested_platform = '${{ inputs.platform }}'
+          requested_arch = '${{ inputs.architecture }}'
+
+          # Get the platforms to build on (set `x86-platforms` and `arm-platforms`)
+          if requested_platform == 'all':
+            x86-platforms = ${{ fromJson(env.ALL_X86_IMAGES) }} if requested_arch == 'all' or requested_arch == 'x86_64' else []
+            arm-platforms = ${{ fromJson(env.ALL_ARM_IMAGES) }} if requested_arch == 'all' or requested_arch == 'arm64'  else []
+          else:
+            if requested_arch == 'all' or requested_arch == 'x86_64':
+              x86-platforms = [requested_platform] if requested_platform in ${{ fromJson(env.ALL_X86_IMAGES) }} else []
+            if requested_arch == 'all' or requested_arch == 'arm64':
+              arm-platforms = [requested_platform] if requested_platform in ${{ fromJson(env.ALL_ARM_IMAGES) }} else []
+
+          # Generate warnings/errors if requested combination is not supported
+          if requested_platform is not 'all':
+            if (requested_arch == 'x86_64' and requested_platform not in x86-platforms) or \
+               (requested_arch == 'arm64'  and requested_platform not in arm-platforms):
+              print(f"::error title='Unsupported Combination'::Requested platform {requested_platform} is not supported on {requested_arch}")
+              exit(1)
+            if requested_arch == 'all':
+              if requested_platform not in x86-platforms + arm-platforms:
+                print(f"::error title='Unsupported Combination'::Requested platform {requested_platform} is not supported on any architecture")
+                exit(1)
+              elif requested_platform not in arm-platforms or requested_platform not in x86-platforms:
+                unsupported_arch = 'arm64' if requested_platform not in arm-platforms else 'x86_64'
+                print(f"::warning title='Unsupported Combination'::Requested platform {requested_platform} is not supported on {unsupported_arch}. Skipping {unsupported_arch}")
```

Comment:
Nice and sophisticated as well... just remember to do manual testing on all the code paths that are not covered by CI (including the invalid configurations)